app/assets/javascripts/uploadcare/settings.coffee

```diff
@@ -201,6 +203,16 @@ uploadcare.namespace 'settings', (ns) ->
 
     if settings.validators
       settings.validators = settings.validators.slice()
+    
+    if settings.previewUrlProvider and not settings.overridePreviewUrl
+      settings.overridePreviewUrl = (url, info) => 
+        useGetParam = /\?$/.test(settings.previewUrlProvider)
+        location = if useGetParam then "url=#{encodeURIComponent(url)}" else "/#{url}"
+        utils.normalizeUrl(settings.previewUrlProvider) + location
```

Comment:
location already contains domain from CDNBase. Does it really should?

Reply:
I think so. Why not?
I think URL-signing backed should know nothing about used CDN base. It's just check auth and sign urls.

Reply:
At least because now preview URL will look like:

`https://preview.base.com/some/path/https://cdn.base.com/uuid/-/opertaions/`

My original idea was to replace cdnBase with previewBase if the least is set:

`https://preview.base.com/some/path/uuid/-/opertaions/`